src/components/forms/VpnServerForm.js

```diff
@@ -0,0 +1,386 @@
+    // removing white space and checking for empty value
+    let value = event.target.value.replace(/\s*/g, "");
+    if (value === "") value = null;
+    this.setState({ client_dns_server_ips: value });
+  }
+
+  handleMultiSelectChange(name, value) {
+    this.setState(this.setNameToValue(name, value));
+  }
+
+  handleToggle(name) {
+    this.setState(this.toggleStateBoolean(name, this.state));
+  }
+
+  render() {
+    let composedId = `vpn-server-form-${this.props.data.name}`;
+    let classNameModalCheck = this.props.isModal
+      ? "fieldWidthSmaller"
+      : "fieldWidth";
+    let routeProps = {
+      invalidCallback: this.props.invalidVpnServerRouteCallback,
+      invalidTextCallback: this.props.invalidVpnServerRouteTextCallback,
+      arrayParentName: this.props.data.name,
+      parent_name: this.props.data.name,
+    };
+    transpose({ ...this.props.vpnServerRouteProps }, routeProps);
+    return (
+      <>
+        <IcseFormGroup>
+          {/* vpn server name */}
+          <IcseNameInput
+            id={composedId}
+            className={classNameModalCheck}
+            componentName="vpn-server-name"
+            value={this.state.name}
+            onChange={this.handleInputChange}
+            invalidCallback={() =>
+              this.props.invalidCallback(this.state, this.props)
+            }
+            invalidText={this.props.invalidTextCallback(this.state, this.props)}
+            hideHelperText
+          />
+          {/* certificate_crn */}
+          <IcseTextInput
+            componentName="vpn-server-certificate-crn"
+            className={classNameModalCheck}
+            tooltip={{
+              content:
+                "Must use Secrets Manager to generate certificate_crn of secret for this VPN server.",
+              align: "top-left",
+            }}
+            id="certificate_crn"
+            labelText="Certificate CRN"
+            value={this.state.certificate_crn}
+            onChange={this.handleInputChange}
+            field="certificate_crn"
+          />
+        </IcseFormGroup>
+        <IcseFormGroup>
+          {/* method toggle */}
+          <IcseSelect
+            formName="vpn-server-method"
+            className={classNameModalCheck}
+            name="method"
+            labelText="Method"
+            groups={["certificate", "username"]}
+            value={this.state.method}
+            handleInputChange={this.handleInputChange}
+          />
+          {/* client_ca_crn or identity_provider */}
+          {this.state.method === "certificate" ? (
+            <IcseTextInput
+              componentName="vpn-server-client-ca-crn"
+              className={classNameModalCheck}
+              id="client_ca_crn"
+              labelText="Client CA CRN"
+              value={this.state.client_ca_crn}
+              onChange={this.handleInputChange}
+              field="client_ca_crn"
+            />
+          ) : (
+            <IcseTextInput
```

Comment:
remove `identity_provider` input, only one value `iam` allowed. not part of issue